### ISSUE TITLE
Proposal: Support Discord Markdown

### ISSUE THREAD

**Issue by l-zeuch** [l-zeuch/yagpdb.vim#23]
### Description

When writing output of custom commands, it is quite a regular thing to use Discord's markdown to spice it up a little and make it more appealing.

### Solution

For this, I suggest to detect and highlight Discord's markdown markers, just like some markdown extensions already do. I understand that some formatting might get a little weird, for this we'd have to think of a way and maybe look laterally, how other extensions specifically for markdown handle that.

Furthermore it might be worthwhile to highlight markdown only when it makes sense, i.e. in `print*`, `cembed`, field values,  descriptions, and of course in plain text.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

**Comment by LRitzdorf** [comment]
Great idea! This would be nice, but I'm thinking it might throw a monkey wrench into some of our existing highlighting, if not done properly. I can take a look at that.

Also, in terms did appropriate areas to perform this highlighting, we should be able to restrict it to normal text and within the String group without too much trouble.

**Comment by l-zeuch** [comment]
> Also, in terms of appropriate areas to perform this highlighting, we should be able to restrict it to normal text and within the String group without too much trouble.

Yes but also kinda no -- note that some functions take string arguments, but don't actually print said argument to chat, for example `dbGet`'s key argument, which is linked to the string hl-group.

This should be considered and was the point I tried to chase in my initial comment, to only highlight markdown where it actually makes sense.

I presume that was what you meant with it might throw a monkey wrench into our existing highlighting?

**Comment by l-zeuch** [comment]
> For example, if we have a Markdown codeblock region, CC syntax elements like cslice might highlight inside that if we aren't careful.

Ah, that does make a little more sense. Totally missed that in the first place :sweat_smile: 

> I know I did a bit of a syntax rewrite earlier, but it still feels a bit messy, so I might try to reevaluate things and add Markdown formatting as part of that.

Sure, that sounds good -- A cleaner code focussed on extensibility is always preferable.
